code/game/objects/structures/beacon.dm

```diff
@@ -0,0 +1,45 @@
+/obj/structure/strangebeacon
+	icon = 'icons/obj/machines/excelsior/objects.dmi'
+	icon_state = "strange_beacon"
+	var/nosignal = 0
+
+/obj/structure/strangebeacon/attack_hand(mob/living/user as mob)
+	if(nosignal == 0)
+		playsound(loc, 'sound/machines/twobeep.ogg', 50, 1)	//Plays a beep
+		nosignal = 1
+
+/obj/structure/strangebeacon/bots //who you gonna call
+/obj/structure/strangebeacon/bots/attack_hand(mob/living/user as mob)
+	if(nosignal == 0)
+		var/counter = 0
+		var/counterfinish = rand(1,2)
+
+		while(counter < counterfinish)
+			counter++
+			new /mob/living/bot/miningonestar/resources/agressive ( locate( get_step(src, pick(NORTH, WEST, EAST, SOUTH) ) ))
+		nosignal = 1
+
+/obj/structure/strangebeacon/pods
+/obj/structure/strangebeacon/pods/attack_hand(mob/living/user as mob)
+	if(nosignal == 0)
+		var/drop_x = src.x-2
+		var/drop_y = src.y-2
+		var/drop_z = src.z
+		var/drop_type = pick(supply_drop_random_loot_types())
+
+		spawn(rand(100,300))
+			new /datum/random_map/droppod/supply(null, drop_x, drop_y, drop_z, supplied_drop = drop_type) // Splat.
```

Comment:
Is it intended that there's zero warning before pods start dropping? Can get pretty frustrating since they can just gib you with unlucky hit.